test/stress_tests/test_many_tasks_and_transfers.py

```diff
@@ -40,31 +40,36 @@ def method(self, size, *xs):
         return np.ones(size, dtype=np.uint8)
 
 
-# Launch a bunch of tasks.
+# Launch a bunch of tasks. (approximately 200 seconds)
 start_time = time.time()
 logger.info("Submitting many tasks.")
 for i in range(10):
     logger.info("Iteration {}".format(i))
     ray.get([f.remote(0) for _ in range(100000)])
 logger.info("Finished after {} seconds.".format(time.time() - start_time))
 
-# Launch a bunch of tasks, each with a bunch of dependencies.
+# Launch a bunch of tasks, each with a bunch of dependencies. TODO(rkn): This
+# test starts to fail if we increase the number of tasks in the inner loop from
+# 500 to 1000. (approximately 615 seconds)
 start_time = time.time()
 logger.info("Submitting tasks with many dependencies.")
 x_ids = []
-for i in range(5):
-    logger.info("Iteration {}".format(i))
-    x_ids = [f.remote(0, *x_ids) for _ in range(10000)]
-ray.get(x_ids)
-logger.info("Finished after {} seconds.".format(time.time() - start_time))
+for _ in range(5):
+    for i in range(20):
+        logger.info("Iteration {}. Cumulative time {} seconds".format(
+            i,
+            time.time() - start_time))
```

Comment:
In general, we should avoid using `string.format` for logging. 
Because when using `string.format`, the message string always needs to be formatted, even when the actual logging level is higher than `info`.
Instead, we should use `logger.info("message %s %s", arg1, arg2)`, which have little overhead when logging level is higher than `info`.
This problem exists in a lot of places in Ray code base. We should avoid it in the future.

Reply:
Ok, sounds reasonable. Do you think we should do this only inside of `logger` or in general?

Reply:
In general, we should avoid the following pattern.
```python
res = heavy_computation()  # do heavy computation to get a result.
func(res)  # pass the result to a function
def func(res):
   if some_condition:
      use_result(res)  # but in the function, the result is only used when certain condition is met.
```
`logger` is the most common use case that have this pattern. If we see other use cases, we should be aware as well.